store/sqlstore/channel_store_categories.go

```diff
@@ -813,6 +814,8 @@ func (s SqlChannelStore) addChannelToFavoritesCategoryT(transaction *gorp.Transa
 	var channel *model.Channel
 	if obj, err := transaction.Get(&model.Channel{}, preference.Name); err != nil {
```

Comment:
I thought we get a `sql.ErrNoRows` if no rows match?

In any case, can we add a test ?

Reply:
From [mattermost/gorp](https://github.com/mattermost/gorp/blob/master/gorp.go):
```golang
err = row.Scan(dest...)
if err != nil {
	if err == sql.ErrNoRows {
		err = nil
	}
	return nil, err
}
```

Anyway, added a couple of test for this.

Reply:
Oh .. this is only for `Get`. We should actually use `SelectOne`. Anyways, no need to change that now.